### PR TITLE
feat: split dev prod resources

### PR DESCRIPTION
**Pull request by bashleigh**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #6581

### CODE REVIEW

File: packages/deployment-service/src/functions/pipeline/pipeline-setup.ts [hunk 1]
```diff
@@ -81,7 +79,7 @@ export const pipelineSetup: SQSHandler = async (event: SQSEvent, context: Contex
               Quantity: 1,
               Items: [`${pipeline.subDomain}.${process.env.ROOT_DOMAIN}`],
             },
-            Comment: `Cloudfront distribution for pipeline [${pipeline.id}]`,
+            Comment: `Cloudfront distribution for pipeline [${pipeline.id}] [${process.env.NODE_ENV === 'production' ? 'prod' : 'dev'}]`,
```
Comment (willmcvay) [inline packages/deployment-service/src/functions/pipeline/pipeline-setup.ts:82]: Should be updated right?

File: packages/deployment-service/src/functions/pipeline/pipeline-setup.ts [hunk 2]
```diff
@@ -134,7 +132,7 @@ export const pipelineSetup: SQSHandler = async (event: SQSEvent, context: Contex
                 },
               },
             ],
-            Comment: `Adding additional A record for pipeline [${pipeline.id}]`,
+            Comment: `Adding additional A record for pipeline [${pipeline.id}]  [${process.env.NODE_ENV === 'production' ? 'prod' : 'dev'}]`,
```
Comment (willmcvay) [inline packages/deployment-service/src/functions/pipeline/pipeline-setup.ts:135]: And here

File: packages/deployment-service/src/functions/pipeline/pipeline-setup.ts (hunk 1 above)
Comment (bashleigh) [reply packages/deployment-service/src/functions/pipeline/pipeline-setup.ts:82]: This is added to the lambda using env prop on lambda creation in cdk setup

File: packages/deployment-service/src/functions/pipeline/pipeline-setup.ts (hunk 2 above)
Comment (bashleigh) [reply packages/deployment-service/src/functions/pipeline/pipeline-setup.ts:135]: same again: This is added to the lambda using env prop on lambda creation in cdk setup

**Review by willmcvay — APPROVED** [review]

**Pull request merged**